lib/UndoInstCombinePass.cpp

```diff
@@ -224,3 +225,30 @@ bool clspv::UndoInstCombinePass::UndoWideVectorShuffleCast(Instruction *inst) {
 
   return true;
 }
+
+bool clspv::UndoInstCombinePass::UndoGEPRemove(Instruction *inst) {
+  auto load = dyn_cast<LoadInst>(inst);
+  if (!load)
+    return false;
+
+  auto gv = dyn_cast<GlobalVariable>(load->getPointerOperand());
+  if (!gv)
+    return false;
+
+  auto value_ty = gv->getValueType();
+  if (auto* value_vec_ty = dyn_cast<VectorType>(value_ty)) {
```

Comment:
My understanding is that LLVM will elide any gep with all 0 indices. The vector case is definitely the most common for clspv.